src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -76,6 +54,65 @@ public HttpApiHandler(DurableTaskExtension config, TraceWriter traceWriter)
             return response;
         }
 
+
+        internal async Task<HttpResponseMessage> CreateCheckStatusResponse(
+            HttpRequestMessage request,
+            string instanceId,
+            OrchestrationClientAttribute attribute,
+            int totalTimeout,
+            int retryTimeout)
+        {
+            GetClientResponseLinks(request, instanceId, attribute, out var statusQueryGetUri, out var sendEventPostUri,
+                out var terminatePostUri);
+
+            if (totalTimeout < retryTimeout)
+            {
+                throw new InvalidOperationException($"Total timeout {totalTimeout} should be bigger than {retryTimeout}");
+            }
+
+            var iterationCount = totalTimeout / retryTimeout;
```

Comment:
I suggest a more robust approach to managing the timeout. The problem with `iterationCount` is that it doesn't work well when `totalTimeout` is not divisible by `retryTimeout`. For example, what should the correct behavior be when `totalTimeout` is `30` but `retryTimeout` is `8`? In the current implementation, the total timeout is actually 24 seconds instead of 30 seconds.